soroban-env-common/src/env.rs

```diff
@@ -146,8 +146,14 @@ macro_rules! call_macro_with_all_host_functions {
                 /// invoked by a contract.
                 {"0", fn get_invoking_contract() -> Object }
                 {"1", fn obj_cmp(a:RawVal, b:RawVal) -> i64 }
-                {"2", fn contract_event(v:RawVal) -> RawVal }
-                {"3", fn system_event(v:RawVal) -> RawVal }
+                /// Records a contract event. `topics` is expected to be a `SCVec` with
+                /// length <= 4 that cannot contain Vecs, Maps, or Binaries > 32 bytes
+                /// On succes, returns an `SCStatus::Ok`.
+                {"2", fn contract_event(topics:Object, data:RawVal) -> RawVal }
+                /// Records a system event. `topics` is expected to be a `SCVec` with
+                /// length <= 4 that cannot contain Vecs, Maps, or Binaries > 32 bytes
+                /// On succes, returns an `SCStatus::Ok`.
+                {"3", fn system_event(topics:Object, data:RawVal) -> RawVal }
```

Comment:
❓ I thought system_event was an internal function, not a function that we'd let a contract call. Is that right?

Reply:
You are right, I missed it from reading the CAP. 
I've fixed it by moving the `system_event` function internal. However, this will fail if called within the `HostFunction` context, i.e. invoked from stellar-core. Added a new issue: https://github.com/stellar/rs-soroban-env/issues/345.